File: python_stack/python/fundamentals/fibonacci_recursive_memoization.js
var fibCache = {}

function fibonacci(n){
    if (n <= 1){
        return n
    }
    if (n in fibCache){
        return fibCache[n]
    }
    else{
        var value = fibonacci(n - 1) + fibonacci(n - 2)
        fibCache[n] = value
        return value
    }
}

console.log(fibonacci(147600000))

function fibonacci2(n){
    if (n <= 1){
        return n
    }
    return(fibonacci2(n - 1) + fibonacci2(n - 2))
}

// console.log(fibonacci2(40))

File: python_stack/python/fundamentals/fibonacci_recursive_memoization.py
fib_cache = {}


def fibonacci(n):
    if n <= 1:
        return n
    if n in fib_cache:
        return fib_cache[n]
    else:
        value = fibonacci(n - 1) + fibonacci(n - 2)
    fib_cache[n] = value
    return value


# print(fibonacci(900))


def fibonacci2(n):
    if n <= 1:
        return n
    else:
        return fibonacci2(n - 1) + fibonacci2(n - 2)

# print(fibonacci2(40))